cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MetricsMessageCallback.java

```diff
@@ -93,6 +97,16 @@ public MetricValues apply(FetchedMessage input) {
     }
   }
 
+  private void addProcessingStats(List<MetricValues> records) {
+    int count = records.size();
+    long now = System.currentTimeMillis();
+    long delay = now - records.get(records.size() - 1).getTimestamp();
```

Comment:
check if records is not empty. the list could be empty if we can't decode the messages we received right ?